### PR TITLE
display less evidences in uniprotkb results

### PR DESCRIPTION
**Pull request by aurel-l**
## Purpose
Remove the evidence tags on the UniProtKB results columns [TRM-25949](https://www.ebi.ac.uk/panda/jira/browse/TRM-25949)

## Approach
Remove the evidence tags on *some* of the UniProtKB results columns, use plain text expandable lists instead.
Will need a complete review of all the columns anyway because a lot are using the same components as on the Entry page when it doesn't really make sense within a table cell. And some columns just don't have an implementation yet.

## Testing
Updated snapshots

## Checklist
- [x] My PR is scoped properly, and “does one thing only”
- [x] I have reviewed my own code
- [x] I have checked that linting checks pass and type safety is respected
- [x] I have checked that tests pass and coverage has at least improved, and if not explained the reasons why

Check the columns that have been modified (see diff view of the column config file)

### CODE REVIEW

File: src/uniprotkb/config/UniProtKBColumnConfiguration.tsx [hunk 1]
```diff
@@ -204,82 +201,68 @@ UniProtKBColumnConfiguration.set(UniProtKBColumn.length, {
...
-              geneData.geneName && (
-                <div key={geneData.geneName.value}>
-                  {geneData.geneName.value}
-                </div>
-              )
-          )}
+
+    const names = geneNamesData?.map((geneNames) => geneNames.geneName?.value);
+
+    return names?.length ? (
```
Comment (dlrice) [inline src/uniprotkb/config/UniProtKBColumnConfiguration.tsx:210]: I see the pattern `predicate ? true-branch : null` for all of the column modifications in this PR - is there a reason this is used rather `predicate && true-branch` (which I personally find a little bit easier to read)?

**Review by dlrice — APPROVED** [review]

File: src/uniprotkb/config/UniProtKBColumnConfiguration.tsx (hunk 1 above)
Comment (aurel-l) [reply src/uniprotkb/config/UniProtKBColumnConfiguration.tsx:210]: This is because these are lengths, so you could end up displaying "0".
I could also test on the array itself but not its length, but I think the Expandable list renders an empty list, but I want it to render nothing, not an empty `<ul>`.

Comment (dlrice) [reply src/uniprotkb/config/UniProtKBColumnConfiguration.tsx:210]: thoughts on?
```
    return (
      Boolean(names?.length) && (
        <ExpandableList descriptionString="gene names" numberCollapsedItems={1}>
          {names}
        </ExpandableList>
      )
    );
```

Comment (aurel-l) [reply src/uniprotkb/config/UniProtKBColumnConfiguration.tsx:210]: I would prefer:
```
return (
      names && (
        <ExpandableList descriptionString="gene names" numberCollapsedItems={1}>
          {names}
        </ExpandableList>
      )
    );
```
And modify ExpandableList in Franklin to return null when the number of children is 0. What do you think?

Comment (dlrice) [reply src/uniprotkb/config/UniProtKBColumnConfiguration.tsx:210]: That sounds good - also names could potentially include `undefined` elements so could determine if this is the case within franklin too

**Review by dlrice — APPROVED** [review]
